services/src/test/java/org/keycloak/vault/PlainTextVaultProviderFactoryTest.java

```diff
@@ -0,0 +1,1300 @@
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.keycloak.Config;
+import org.keycloak.common.enums.SslRequired;
+import org.keycloak.component.ComponentModel;
+import org.keycloak.models.AuthenticationExecutionModel;
+import org.keycloak.models.AuthenticationFlowModel;
+import org.keycloak.models.AuthenticatorConfigModel;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.ClientScopeModel;
+import org.keycloak.models.GroupModel;
+import org.keycloak.models.IdentityProviderMapperModel;
+import org.keycloak.models.IdentityProviderModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.OTPPolicy;
+import org.keycloak.models.PasswordPolicy;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.RequiredActionProviderModel;
+import org.keycloak.models.RequiredCredentialModel;
+import org.keycloak.models.RoleModel;
+import org.keycloak.services.DefaultKeycloakSession;
+import org.keycloak.services.DefaultKeycloakSessionFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Tests for {@link PlainTextVaultProviderFactory}.
+ *
+ * @author Sebastian Łaskawiec
+ */
+public class PlainTextVaultProviderFactoryTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void shouldInitializeVaultCorrectly() {
+        //given
+        VaultConfig config = new VaultConfig(Scenario.EXISTING.getAbsolutePathAsString(), Boolean.FALSE);
+        PlainTextVaultProviderFactory factory = new PlainTextVaultProviderFactory();
+
+        KeycloakSession session = new DefaultKeycloakSession(new DefaultKeycloakSessionFactory());
+        session.getContext().setRealm(new VaultRealmModel());
+
+        //when
+        factory.init(config);
+        VaultProvider provider = factory.create(session);
+
+        //then
+        assertNotNull(provider);
+    }
+
+    @Test
+    public void shouldThrowAnExceptionWhenTryingToCreateProviderOnDisabledFactory() {
+        //given
+        VaultConfig config = new VaultConfig(Scenario.EXISTING.getAbsolutePathAsString(), Boolean.TRUE);
+        PlainTextVaultProviderFactory factory = new PlainTextVaultProviderFactory();
+
+        expectedException.expect(IllegalStateException.class);
+
+        //when
+        factory.init(config);
+        factory.create(null);
+
+        //then - verified by the ExpectedException rule
+    }
+
+    @Test
+    public void shouldThrowAnExceptionWhenUsingNonExistingDirectory() {
+        //given
+        VaultConfig config = new VaultConfig(Scenario.NON_EXISTING.getAbsolutePathAsString(), Boolean.FALSE);
+        PlainTextVaultProviderFactory factory = new PlainTextVaultProviderFactory();
+
+        expectedException.expect(VaultNotFoundException.class);
+
+        //when
+        factory.init(config);
+
+        //then - verified by the ExpectedException rule
+    }
+
```

Comment:
Let's consider the latter parameter first - it's a boolean flag indicates that the provider is disabled. it's a `Boolean`, so it can be `true`, `false` or `null`:

* `true` is tested in `shouldThrowAnExceptionWhenTryingToCreateProviderOnDisabledFactory`.
* `false` is tested in `shouldInitializeVaultCorrectly`.
* `null` is tested in `shouldInitializeVaultCorrectlyWithNullDisabledFlag` (I've just added it). This one is a bit silly as someone would have to explicitly put `null` in a `boolean` configuration param. But I decided to handle it rather than throw a `NullPointerException` from autoboxing.

The second parameter is a `String`, so it can be `null`, represent proper path and a non-existing path:

* `null` is tested in `shouldThrowAnExceptionWhenWithNullDirectory` (just added). This one is tricky since we can not tell if someone misconfigured the provider, or it's just not configured. So the only way to check it, is to try to create a `PlainTextVaultProvider`. If we get an exception, that's a misconfiguration. In an unconfigured case, `PlainTextVaultProviderFactory#create` would never be called.
* existing path is tested in `shouldInitializeVaultCorrectly`
* non-existing path is tested in `shouldThrowAnExceptionWhenUsingNonExistingDirectory`.

Reply:
This one should also be fixed now.